javav2/example_code/redshift/src/main/java/com/example/redshift/scenario/RedshiftActions.java

```diff
@@ -0,0 +1,506 @@
+                    Iterator<JsonNode> iter = rootNode.iterator();
+                    return iter;
+                } catch (IOException e) {
+                    throw new RuntimeException("Failed to read or parse JSON file: " + e.getMessage(), e);
+                }
+            }).thenCompose(iter -> processNodesAsync(clusterId, databaseName, userName, iter, number))
+            .whenComplete((result, exception) -> {
+                if (exception != null) {
+                    logger.info("Error {} ", exception.getMessage());
+                } else {
+                    logger.info("{} records were added to the Movies table." , result);
+                }
+            });
+    }
+
+    private CompletableFuture<Integer> processNodesAsync(String clusterId, String databaseName, String userName, Iterator<JsonNode> iter, int number) {
+        return CompletableFuture.supplyAsync(() -> {
+            int t = 0;
+            try {
+                while (iter.hasNext()) {
+                    if (t == number)
+                        break;
+                    JsonNode currentNode = iter.next();
+                    int year = currentNode.get("year").asInt();
+                    String title = currentNode.get("title").asText();
+
+                    // Use SqlParameter to avoid SQL injection.
+                    List<SqlParameter> parameterList = new ArrayList<>();
+                    String sqlStatement = "INSERT INTO Movies VALUES( :id , :title, :year);";
+                    SqlParameter idParam = SqlParameter.builder()
+                        .name("id")
+                        .value(String.valueOf(t))
+                        .build();
+
+                    SqlParameter titleParam = SqlParameter.builder()
+                        .name("title")
+                        .value(title)
+                        .build();
+
+                    SqlParameter yearParam = SqlParameter.builder()
+                        .name("year")
+                        .value(String.valueOf(year))
+                        .build();
+                    parameterList.add(idParam);
+                    parameterList.add(titleParam);
+                    parameterList.add(yearParam);
+
+                    ExecuteStatementRequest insertStatementRequest = ExecuteStatementRequest.builder()
+                        .clusterIdentifier(clusterId)
+                        .sql(sqlStatement)
+                        .database(databaseName)
+                        .dbUser(userName)
+                        .parameters(parameterList)
+                        .build();
+
+                    getAsyncDataClient().executeStatement(insertStatementRequest);
+                    logger.info("Inserted: " + title + " (" + year + ")");
+                    t++;
+                }
+            } catch (RedshiftDataException e) {
+                throw new RuntimeException("Error inserting data: " + e.getMessage(), e);
+            }
```

Comment:
I'd like to see a more specific error code evaluated here. E.g. 
```suggestion
      } catch (RedshiftDataException e) {
    String errorCode = e.getErrorCode();  // Assume `getErrorCode()` returns the specific Redshift error code
    switch (errorCode) {
        case "400":
            throw new RuntimeException("Invalid input: " + e.getMessage(), e);
        case "500":
            throw new RuntimeException("Internal server error: Redshift service might be down, please retry. " + e.getMessage(), e);
        case "42P01":  // Table not found error in Redshift (example code)
            throw new RuntimeException("Table does not exist: Please check the table name in the query. " + e.getMessage(), e);
        case "23505":  // Duplicate key error in Redshift (example code)
            throw new RuntimeException("Duplicate entry detected: " + e.getMessage(), e);
        default:
            throw new RuntimeException("Error inserting data: " + e.getMessage(), e);
    }
}
```

Reply:
This bloats example. If any of these are thrown, it will be captured in calling code.